source/common/buffer/buffer_impl.cc

```diff
@@ -86,7 +86,10 @@ uint64_t OwnedImpl::length() const { return evbuffer_get_length(buffer_.get());
 
 void* OwnedImpl::linearize(uint32_t size) {
   ASSERT(size <= length());
-  return evbuffer_pullup(buffer_.get(), size);
+  void* const ret = evbuffer_pullup(buffer_.get(), size);
+  RELEASE_ASSERT(ret != nullptr || size == 0,
```

Comment:
Do we want to fail on size == 0 here? I don't feel strongly but it seems like they should be OK?

Reply:
This is explicitly not failing if size == 0 because this case occurs in the tests. And I don't think we should crash when you have something that evaluates to:

```
void* foo = buf.linearize(0);
for (int i = 0; i < 0; i++) {
   x = foo[i];
}
```

Reply:
Sorry, I read this as the reverse that it would crash when size == 0. Yes this makes sense.